src/plugins/lua/lua.cpp

```diff
@@ -0,0 +1,272 @@
+ *
+ * \copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#ifndef SWIG_TYPE_TABLE
+# error Build system error, SWIG_TYPE_TABLE is not defined
+#endif
+
+#include "lua.hpp"
+#ifndef HAVE_KDBCONFIG
+# include <kdbconfig.h>
+#endif
+#include <kdbhelper.h>
+
+#include <key.hpp>
+#include <keyset.hpp>
+#include <libgen.h>
+
+#include <lua.hpp>
+extern "C" {
+#include <lualib.h>
+}
+#include "runtime.h"
+
+using namespace ckdb;
+#include <kdberrors.h>
+
+static void Lua_fromSWIG(lua_State *L, ckdb::Key *key)
+{
+	swig_type_info *ti = SWIG_TypeQuery(L, "kdb::Key *");
+	if (key == NULL || ti == NULL)
+		lua_pushnil(L);
+	else
+		SWIG_NewPointerObj(L, new kdb::Key(key), ti, 0);
+}
+
+static void Lua_fromSWIG(lua_State *L, ckdb::KeySet *keyset)
+{
+	swig_type_info *ti = SWIG_TypeQuery(L, "kdb::KeySet *");
+	if (keyset == NULL || ti == NULL)
+		lua_pushnil(L);
+	else
+		SWIG_NewPointerObj(L, new kdb::KeySet(keyset), ti, 0);
+}
+
+extern "C"
+{
+
+typedef struct
+{
+	lua_State *L;
+	int printError;
+	int shutdown;
+} moduleData;
+
+static void *Lua_alloc(void *ud ELEKTRA_UNUSED, void *ptr,
+		size_t osize ELEKTRA_UNUSED, size_t nsize)
+{
+	if (nsize == 0) {
+		elektraFree(ptr);
+		return NULL;
+	}
+	return (elektraRealloc(&ptr, nsize) < 0) ? NULL : ptr;
+}
+
+static void Lua_Shutdown(lua_State *L)
+{
+	if (L)
+		lua_close(L);
+}
+
+static int Lua_Require(lua_State *L, const char *name)
+{
+	lua_getglobal(L, "require");
+	lua_pushstring(L, name);
+	int status = lua_pcall(L, 1, 1, 0);
+	if (status == LUA_OK)
+		lua_setglobal(L, name);
+	return status;
+}
+
+static int Lua_CallFunction_Int(lua_State *L, int nargs, ckdb::Key *errorKey)
+{
+	int ret = -1;
+	if (lua_pcall(L, nargs, 1, 0) != LUA_OK)
+		ELEKTRA_SET_ERROR(125, errorKey, lua_tostring(L, -1));
+	else
+	{
+		if (!lua_isnumber(L, -1))
+			ELEKTRA_SET_ERROR(125, errorKey,
+					"Return value is no integer");
+		else
+			ret = lua_tonumber(L, -1);
+	}
+	return ret;
+}
+
+static int Lua_CallFunction_Helper1(lua_State *L, const char *funcName,
+	ckdb::Key *errorKey)
+{
+	int ret = 0;
+	int top = lua_gettop(L);
+	lua_getglobal(L, funcName);
+	if (lua_isfunction(L, -1))
+	{
+		Lua_fromSWIG(L, errorKey);
+		ret = Lua_CallFunction_Int(L, 1, errorKey);
+	}
+	lua_settop(L, top);
+	return ret;
+}
+
+static int Lua_CallFunction_Helper2(lua_State *L, const char *funcName,
+	ckdb::KeySet *returned, ckdb::Key *parentKey)
+{
+	int ret = 0;
+	int top = lua_gettop(L);
+	lua_getglobal(L, funcName);
+	if (lua_isfunction(L, -1))
+	{
+		Lua_fromSWIG(L, returned);
+		Lua_fromSWIG(L, parentKey);
+		ret = Lua_CallFunction_Int(L, 2, parentKey);
+	}
+	lua_settop(L, top);
+	return ret;
+}
+
+int elektraLuaOpen(ckdb::Plugin *handle, ckdb::Key *errorKey)
+{
+	KeySet *config = elektraPluginGetConfig(handle);
+	Key *script = ksLookupByName(config, "/script", 0);
+	if (script == NULL)
+		return 0; // success if no script to execute
+
+	if (keyString(script) == NULL)
```

Comment:
Yes, /module is a special marker saying: don't expect any configuration. The plugin will be below system/elektra/modules and should return some standard contract. Its explained a bit in plugin interface, see elektraDocOpen.

Your code looks good now.